### PR TITLE
tlclient: init at 4.12.0

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/105399

###### Things done
Packaged non-free thin linc client for linux x86_64

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NomisIV** [comment]
I don't know what it's worth, but this has got it working for me. I copied the default.nix to my machine and imported it using `pkgs.callPackage`, and it just works! Thank you very much!

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:14]: I would prefer that you use `mkDesktopItem` instead and created a proper desktop item for NixOS. Using `copyDesktopItems` can also help adding the desktop item to your derivation without copying the generated desktop file manually. See here for details: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/X11/opentabletdriver/default.nix

If you're going to use like this, at least pin to a commit so we avoid breakage if the desktop file in Arch changes: https://aur.archlinux.org/cgit/aur.git/plain/tlclient.desktop?h=tlclient&id=fd126edd54a012caa8c9ce47ab8d7fb25e982e74

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:18]: Same here, pin to a commit instead: https://aur.archlinux.org/cgit/aur.git/plain/tlclient.png?h=tlclient&id=fd126edd54a012caa8c9ce47ab8d7fb25e982e74

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [reply pkgs/applications/networking/remote/tlclient/default.nix:14]: BTW, checking the desktop file, `Exec=/usr/bin/tlclient` wouldn't work in NixOS I think since this is not a valid path to the binary. Also there is a `Version=4.10.0` that needs to be bumped at every upgrade (so it is wrong right now), but either way I think this information about version is irrelevant.

So my recommendation is to really use `mkDesktopItem`.

Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix]: Fixed. BTW I didn't know such tool existed, thanks !

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix]: Fixed.

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:44]: It is missing the desktop item, I think because you forgot to add the `runHook preInstall` at the start of `installPhase` and `runHook postInstall` at the end of it.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 106052` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tlclient</li>
  </ul>
</details>

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [reply pkgs/applications/networking/remote/tlclient/default.nix:44]: Without it, `copyDesktopItems` hook will not run since it runs at `postInstall` phase (I think).

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:66]: Asking here, why not add yourself to `maintainers.nix`?

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix]: Fixed.

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:33]: FYI, it is recommended that you add both `preInstall` and `postInstall` hooks when overwriting phases: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md. This way, it avoids mistakes if someones needs to add a `preInstall` hook for example.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 106052` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tlclient</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [reply pkgs/applications/networking/remote/tlclient/default.nix:33]: But I am approving since for now it is looking right (the desktop file is included).

Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix:33]: Neat tool !

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:38]: This seems misindented,

Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix]: Fixed.

**Review by thiagokokada — APPROVED** [review]

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/networking/remote/tlclient/default.nix:6]:
```suggestion
, fetchurl
, autoPatchelfHook
, copyDesktopItems
, makeDesktopItem
, alsaLib
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 106052` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>tlclient</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (NomisIV) [inline pkgs/applications/networking/remote/tlclient/default.nix:19]: This can now be updated to 4.12.1-6733

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (NomisIV) [inline pkgs/applications/networking/remote/tlclient/default.nix:24]: 04pnw13vj04lxkfq4vvjsxy1lj6sb5wsivf51rmb349b4g0g9fb5

**Review by NomisIV — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix]: Fixed, it's now 4.13.0

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (SCOTT-HAMILTON) [reply pkgs/applications/networking/remote/tlclient/default.nix]: Fixed.

**Comment by thiagokokada** [comment]
Please target `master`, `staging` is mostly for mass rebuilds.

And sorry for your PR being lost, after you retarget it to `master` I will do one final test and can merge this PR if everything is right.

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:7]:
```suggestion
, alsa-lib
```

File: pkgs/applications/networking/remote/tlclient/default.nix
Comment (thiagokokada) [inline pkgs/applications/networking/remote/tlclient/default.nix:29]:
```suggestion
  propagatedBuildInputs = [ alsa-lib libX11 pcsclite ];
```

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 106052` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tlclient</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Please retarget your PR to `master` instead of `staging`.

**Review by jonringer — APPROVED** [review]
diff LGTM
gui launches fine

```
https://github.com/NixOS/nixpkgs/pull/106052

1 package built:
tlclient
```

This can go in master though. But everyone else looks good :+1:

**Pull request closed without merging**
